### ISSUE TITLE
Add something similar to dash:closedByTypes

### ISSUE THREAD

**Issue by HolgerKnublauch** [w3c/shacl#43]
sh:closed is limited to the property shapes at a single node shape, but it does not consider properties that are "inherited" from superclasses.

dash:closedByTypes (https://datashapes.org/constraints.html#ClosedByTypesConstraintComponent) is a solution to that and could be lifted into the SHACL Core vocab. Alternatively, we could add a flag sh:closedByTypes to sh:closed so that it will also walk up the superclasses.

**Comment by bergos** [comment]
It should also cover `sh:node`. Not sure if it can be the same flag or if someone wants to distinguish between class and node.

Regarding how the flag could be defined, maybe we can allow non-boolean values like `sh:ByType`, `sh:ByNode`, or something like that. Would that count as a breaking change?

**Comment by HolgerKnublauch** [comment]
Yes, good point: sh:node is similar to rdfs:subClassOf and belongs into the same category.
